File: libwebrtc-sys/src/async_dns_resolver_factory.cc
#include "async_dns_resolver_factory.h"

std::unique_ptr<webrtc::AsyncDnsResolverFactoryInterface> create_arcas_cxx_async_dns_resolver_factory()
{
    webrtc::BasicAsyncResolverFactory fact;
    auto blah = fact.Create();
    auto wrapped = std::make_unique<webrtc::BasicAsyncResolverFactory>();
    blah = wrapped->Create();
    auto result = std::make_unique<webrtc::WrappingAsyncDnsResolverFactory>(std::move(wrapped));
    return std::move(result);
}

File: libwebrtc-sys/src/async_dns_resolver_factory.rs
#[cxx::bridge]
pub mod ffi {

    unsafe extern "C++" {
        include!("include/async_dns_resolver_factory.h");

        #[namespace = "webrtc"]
        type AsyncDnsResolverFactoryInterface;

        fn create_arcas_cxx_async_dns_resolver_factory(
        ) -> UniquePtr<AsyncDnsResolverFactoryInterface>;
    }
}
